clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -1895,6 +1900,169 @@ private void recordRegionHybridConsumptionStats(
+   * 2. leader was completed and
+   * 3. the last update time was within 5 seconds
+   */
+  @Override
+  protected boolean checkAndLogIfLagIsAcceptableForHybridStore(
+      PartitionConsumptionState pcs,
+      long offsetLag,
+      long offsetThreshold,
+      boolean shouldLogLag,
+      boolean isOffsetBasedLag,
+      long latestConsumedProducerTimestamp) {
+    boolean isLagAcceptable = offsetLag <= offsetThreshold;
+
+    if (isLagAcceptable && isHybridFollower(pcs)) {
+      // non AA stores have issues reading HB SOS from the RT leading to the standby replicas waiting for
+      // leader completion state header indefinitely, so disabling it until that issue is resolved
+      if (!(getServerConfig().isLeaderCompleteStateCheckEnabled() && this.isActiveActiveReplicationEnabled())) {
+        isLagAcceptable = true;
+      } else if (!pcs.isFirstHeartBeatSOSReceived()) {
+        // wait for the first HB to know if the leader supports sending LeaderCompleteState or not
+        isLagAcceptable = false;
+      } else if (pcs.getLeaderCompleteState().equals(LEADER_COMPLETE_STATE_UNKNOWN)) {
+        // if the leader don't support LeaderCompleteState
+        isLagAcceptable = true;
+      } else {
+        // check if the leader is completed and the last update time was within the configured time
+        isLagAcceptable = pcs.isLeaderCompleted()
+            && ((System.currentTimeMillis() - pcs.getLastLeaderCompleteStateUpdateInMs()) <= getServerConfig()
+                .getLeaderCompleteStateCheckValidIntervalMs());
+      }
+    }
+
+    if (shouldLogLag) {
+      String lagLogFooter;
+      if (isHybridFollower(pcs)) {
+        lagLogFooter = ". Leader Complete State: {" + pcs.getLeaderCompleteState().toString()
+            + "}, Last update In Ms: {" + pcs.getLastLeaderCompleteStateUpdateInMs() + "}.";
+      } else {
+        lagLogFooter = "";
+      }
+      LOGGER.info(
+          "{} [{} lag] partition {} is {}lagging. {}Lag: [{}] {} Threshold [{}]{}",
+          isOffsetBasedLag ? "Offset" : "Time",
+          consumerTaskId,
+          pcs.getPartition(),
+          (isLagAcceptable ? "not " : ""),
+          (isOffsetBasedLag ? "" : "The latest producer timestamp is " + latestConsumedProducerTimestamp + ". "),
+          offsetLag,
+          (isLagAcceptable ? "<" : ">"),
+          offsetThreshold,
+          lagLogFooter);
+    }
+
+    return isLagAcceptable;
+  }
+
+  /**
+   * HeartBeat SOS messages carry the leader completion state in the header. This function extracts the leader completion
+   * state from that header and updates the {@param partitionConsumptionState} accordingly. <p>
+   * If there is no leader completion state header, reset the leader completion state to
+   * {@link LeaderCompleteState#LEADER_COMPLETE_STATE_UNKNOWN} as the leader don't know about this header
+   * (using old version of the code) or the leader may have rolled back to a version that doesn't support this header or
```

Comment:
What if Venice Server rolled back to a version, which doesn't emit HB messages at all, so will the function handle that?

Reply:
This won't handle that and will keep on waiting for the 1st HB. But as all the servers are on versions that sends HB now, should we also handle this scenario?

Reply:
Have similar concern: https://github.com/linkedin/venice/pull/775#discussion_r1419741991

Reply:
Lets go with 1st option as per Min's comment.